Sources/Build/Triple.swift

```diff
@@ -0,0 +1,114 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+/// Triple - Helper class for working with Destination.target values
+///
+/// Used for parsing values such as x86_64-apple-macosx10.10 into
+/// set of enums. For os/arch/abi based conditions in build plan.
+///
+/// @see Destination.target
+public struct Triple {
+    public let tripleString: String
+
+    public let arch: Arch
+    public let vendor: Vendor
+    public let os: OS
+    public let abi: ABI
+
+    public enum Error: Swift.Error {
+        case badFormat
+        case unknownArch
+        case unknownOS
+    }
+
+    public enum Arch: String {
+        case x86_64
+        case armv7
+    }
+
+    public enum Vendor: String {
+        case unknown
+        case apple
+    }
+
+    public enum OS: String {
+        case darwin
+        case macOS = "macosx"
```

Comment:
Not sure why we need separate enums for darwin and macOS

Reply:
This class is inspired by https://github.com/apple/swift-llvm/blob/stable/include/llvm/ADT/Triple.h#L152
And there macOS and Darwin is separated

Reply:
I think it is made for backward compatibility and maybe not necessary for swift
But not sure 
So leave it as is

Reply:
```
➜  ~  `xcrun --find swift` --version
Apple Swift version 4.0 (swiftlang-900.0.65 clang-900.0.37)
Target: x86_64-apple-macosx10.9
➜  ~  `xcrun --find clang` --version
Apple LLVM version 9.0.0 (clang-900.0.37)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

As I see both are still used

Reply:
Yeah, unfortunately clang and swift have separate notions, for... lame historical reasons.

Reply:
(it is also critical to keep them separate because the version numbering scheme differs... the Darwin # comes from the kernel, the macOS number is the actual OS version number)